samples/sensor/qdec/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Valerio Setti <valerio.setti@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/printk.h>
+
+void main(void)
+{
+	struct sensor_value val;
+	int rc;
+	const struct device *dev = DEVICE_DT_GET(DT_ALIAS(qdec0));
+
+	if (!device_is_ready(dev)) {
+		printk("Qdec device is not ready\n");
+		return;
+	}
+
+	printk("Quadrature decoder sensor test\n");
+
+	while (true) {
+		rc = sensor_sample_fetch(dev);
+		if (rc != 0) {
+			printk("Failed to fetch sample (%d)\n", rc);
+			return;
+		}
+
+		rc = sensor_channel_get(dev, SENSOR_CHAN_ROTATION, &val);
```

Comment:
note: this sample is now generic to any kind of sensor that is able to report a rotation channel in degrees, qdec devices are a typical case but we could have others. Maybe rename sample folder?

Reply:
Ok, I understand your point, but I don't have any good idea about a possible naming for such a generic folder. Do you have any suggestion as well? What about "rotation"?

PS = increasing the scope of this sample application will cause changes also to the alias used in the DT, since `qdec` will no more be a good choice

Reply:
Let's keep the `qdec` naming for now, fair enough.